vanilla_first_setup/defaults/user.py

```diff
@@ -143,6 +152,29 @@ def __on_username_entry_changed(self, *args):
             self.__verify_continue()
             self.username = _input
 
+    def __generate_username_from_fullname(self):
+        if self.username_entry.get_text() != '' and self.username_entry.get_text() != self.automatic_username:
+            return
+
+        username = self.fullname_entry.get_text()
+
+        if re.findall('[^ ]+', username) == []: # i.e. just spaces and nothing else (causes the app to crah)
+            return
+
+        username = re.sub(r' ', '_', username)
+        username = re.sub(r'\W+', '', username)
+        username = username.lower()
+
+        current_position = 0
+        while '_' in username:
```

Comment:
Am I missing something here or is this supposed to be the same as `a.replace("_", "")`?